lib/ruby_smb/smb2/packet/ioctl_response.rb

```diff
@@ -17,7 +17,7 @@ class IoctlResponse < RubySMB::GenericPacket
         uint32        :output_count,        label: 'Output Count'
         uint32        :flags,               label: 'Flags'
         uint32        :reserved2,           label: 'Reserved Space'
-        string        :buffer,              label: 'Input Buffer'
+        string        :buffer,              label: 'Input Buffer',        length: -> { output_count }
```

Comment:
You might want to add the `:read_length` parameter to avoid issue when reading:
```ruby
       string        :buffer,              label: 'Input Buffer', read_length: -> { input_count + output_count }
```